static/specification.html

```diff
@@ -59,384 +58,130 @@
-              a non-zero amount, the provider notifies the browser:
-              <ul>
-                <li>
-                  The <a>user agent</a> sets
-                  <code>document.monetization.state</code> to
-                  <code>started</code>.
-                </li>
-                <li>
-                  This MUST occur before the `start` event is fired.
-                </li>
-                <li>
-                  This event is a {{ MonetizationStartEvent }}
-                  containing the Payment Setup Endpoint and the Monetization ID
-                  as properties.
-                </li>
-                <li>
-                  The <a>user agent</a> also emits a {{ progress }} event from
-                  <code>document.monetization</code>, corresponding to this
-                  first packet.
-                </li>
-              </ul>
-            </li>
-            <li>
-              Payment continues for the lifetime of a given meta tag (or
-              semantically equivalent)
-              <ul>
-                <li>
-                  The provider MAY decide to stop/start payment at any time,
-                  e.g. if the user is idle, backgrounds the page, or instructs
-                  the browser to do so.
-                </li>
-                <li>
-                  If the [[[STREAM]]] connection is severed, the provider will
-                  redo the SPSP query to the same Payment Setup Endpoint as
-                  before with the same Monetization ID. The <a>user agent</a>
-                  MUST NOT re-process the `&lt;meta&gt;` Tags Set.
-                </li>
-                <li>
-                  Each time a packet with a non-zero amount is fulfilled, the
-                  provider notifies the browser, and the browser emits a
-                  {{ MonetizationProgressEvent }}
-                  on
-                  <code>document.monetization</code>. The event has details of
-                  the packet.
-                </li>
-                <li>
-                  When a stream is closed the
-                  <code>document.monetization.state</code> MUST be set back to
-                  'pending'
-                </li>
-              </ul>
-            </li>
-          </ul>
-        </li>
-      </ol>
-      <h3 id="payment-handler-flow">Payment Handler Flow</h3>
       <p>
-        A provider can be implemented as a Payment Handler supporting the
-        'webmonetization' payment method (The payment method specification for
-        this payment method is still under development as it is dependant on
-        changes to the Payment Handler API that support streaming payments).
-        Communication between the browser and the provider would use this flow.
+        The monetization keyword indicates a [=payment end-point=] used to
+        monetize the document.
+      </p>
+      <p>
+        ...Fetching and processing details TBW...
```

Comment:
What sort of changes are we anticipating to the flow?

Reply:
I'm basically planning on transplanting: https://html.spec.whatwg.org/#link-type-manifest and adapting it to this. 

That part of HTML handles the fetching and processing and eventually will call back into here (where anything "flow" specific will take over). 

Other changes will be just hooking into the DOM/HTML eventing model. I'll try to have a draft of that this coming week or next.

Reply:
I am not sure that the manifest link type actually matches how we use the tag
* We request the resource (payment endpoint) more than once per document load
* We sometimes do not even request the resource, rather send it to the backend to be used in a potentially async manner. 
* This makes assumptions about how how much the browser will actually implement vs any potential WM Provider plugins ( perhaps implemented as extensions as per @sidvishnoi ) that may want access, for whatever reason, to the payment endpoint url. 

I think we want something closer to a dumber (metadata) tag, like "<link rel='author' >"  that simply encodes a relationship, rather than something that implies a (read: one) fetch,  as per  https://html.spec.whatwg.org/#external-resource-link

> * Links to external resources
These are links to resources that are to be used to augment the current document, generally automatically processed by the user agent. All [external resource links](https://html.spec.whatwg.org/#external-resource-link) have a [fetch and process the linked resource](https://html.spec.whatwg.org/#fetch-and-process-the-linked-resource) algorithm which describes how the resource is obtained.

Reply:
@sublimator - The choice of a `link` tag was very intentional based on feedback from browser vendors and is supposed to mimic the behaviour of other linked resources that are fetched.

I think we need to review the lifecycle of the link and the events that are emitted to make sure we're all on the same page as to what is intended and what developers expect.

Maybe the right way to think about this is to have two separate events and be very clear to developers what these mean:
1. Load - the browser has successfully fetched the payment pointer and was able to do whatever processing is necessary to start sending payments if the WM provider wants to send payments. This event will not fire if the user has disabled monetization either globally or for this origin 
2. Monetization - a payment was sent to the payment pointer

Reply:
To be clear, I'm not against a link tag, but the way we are using it. 

At this point it seems completely obvious to some (I think you and Alex are on the same page here *) that the Browser should be the only component ever to have access to the payment pointer. Only the processed link connection details should be shared with other components. 

Uncompromising on privacy, baking in the tradeoffs.  

With that, sure, why not a 'load' event?

If we are going to have a 'load' event, and it means processing the payment pointer, could we emit it as early as possible? As soon as the payment pointer seems well formed, and the connection details requests are about to be initiated? 

I have a bad habit of jumbling in a lot of different points, but here we go: 
I once asked why we (Coil) don't just do away with WebSockets/BTP and set up a simple http payment endpoint, that could be used for batch payments and tipping, authenticated with privacy pass tokens. Apparently, with the amounts involved with tipping, KYC/AML is more of an issue. That answer could/should be revisited, but I think it's worth considering.

Reply:
> If we are going to have a 'load' event, and it means processing the payment pointer, could we emit it as early as possible? As soon as the payment pointer seems well formed, and the connection details requests are about to be initiated?

I think that would:
1.  Make it future compatible with sharing the payment pointer outside of the Browser/UA if that tradeoff is ever revisited 
2.  Make it so that websites don't have to wait precious milliseconds before making decisions based on the load event, (which does NOT necessarily guarantee payment anyway)

Reply:
> If we are going to have a 'load' event, and it means processing the payment pointer, could we emit it as early as possible? As soon as the payment pointer seems well formed, and the connection details requests are about to be initiated?

I don't see why not.

Reply:
> The choice of a link tag was very intentional based on feedback from browser vendors

It's interesting. I also talked to people on the Chrome payments team who said that meta tags might actually be better. I guess it really depends how you ask the questions.